chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,843 @@
+                    TokensError::IssueErrorIncorrectTicker(_, _)
+                ))
+            ))
+        ));
+
+        // Ticker contain non alpha-numeric byte
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: "💖".as_bytes().to_vec(),
+            amount_to_issue: Amount::from_atoms(52292852472),
+            number_of_decimals: 1,
+            metadata_uri: b"https://some_site.meta".to_vec(),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorIncorrectTicker(_, _)
+                ))
+            ))
+        ));
+
+        // Issue amount is too low
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: "SOME".as_bytes().to_vec(),
+            amount_to_issue: Amount::from_atoms(0),
+            number_of_decimals: 1,
+            metadata_uri: b"https://some_site.meta".to_vec(),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorIncorrectAmount(_, _)
+                ))
+            ))
+        ));
+
+        // Too many decimals
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: "SOME".as_bytes().to_vec(),
+            amount_to_issue: Amount::from_atoms(123456789),
+            number_of_decimals: 123,
+            metadata_uri: b"https://some_site.meta".to_vec(),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorTooManyDecimals(_, _)
+                ))
+            ))
+        ));
+
+        // URI is too long
+        let values = vec![OutputValue::Token(TokenData::TokenIssuanceV1 {
+            token_ticker: b"SOME".to_vec(),
+            amount_to_issue: Amount::from_atoms(52292852472),
+            number_of_decimals: 1,
+            metadata_uri: "https://some_site.meta".repeat(1024).as_bytes().to_vec(),
+        })];
+        assert!(matches!(
+            process_token(&mut test_framework, ParentBlock::BestBlock, values),
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::TokensError(
+                    TokensError::IssueErrorIncorrectMetadataURI(_, _)
+                ))
+            ))
+        ));
+    });
```

Comment:
Always when you test something that doesn't work, you end it with something that works so that you have a reference. Here, you tested multiple cases that don't work. You should finally test passing the working case with everything being valid.

Reply:
I have a valid case at the start of the test. However, I fixed the comment to highlight valid case.